README.md

```diff
@@ -529,6 +529,86 @@ If instead your Postgres type is an array of varchar or text, use the following
 import com.augustnagro.magnum.pg.enums.PgStringToScalaEnumSqlArrayCodec
 ```
 
+#### Json, JsonB, XML
+
+You can map `json`, `jsonb`, and `xml` columns to Scala classes by implementing `JsonDbCodec`, `JsonBDbCodec`, and `XmlDbCodec` (respectively).
+
+As an example, assume we have table `car`:
+
+```sql
+CREATE TABLE car (
+  id bigint primary key,
+  last_service json not null
+);
+```
+
+And `last_service` looks like:
+
+```json
+{"mechanic": "Bob", "date":  "2024-05-04"}
+```
+
+We can model the relation in Scala with:
+
+```scala
+@Table(PostgresDbType, SqlNameMapper.CamelToSnakeCase)
+case class Car(
+    @Id id: Long,
+    lastService: LastService
+) derives DbCodec
+
+case class LastService(mechanic: String, date: LocalDate)
+```
+
+However, this won't compile because we're missing a given `DbCodec[LastService]`. To get there, first we have to pick a Scala JSON library. Nearly all of them support creating derived codecs; the example below shows how it's done in [Play Json](https://github.com/playframework/play-json):
+
+```scala
+case class LastService(mechanic: String, date: LocalDate)
+
+object LastService:
+  given OFormat[LastService] = Json.format[LastService]
+```
+
+Next, we should extend `JsonDbCodec` to implement our own `PlayJsonDbCodec`:
+
+```scala
+import com.augustnagro.magnum.pg.json.JsonDbCodec
+import play.api.libs.json.*
+
+trait PlayJsonDbCodec[A] extends JsonDbCodec[A]
+
+object PlayJsonDbCodec:
+  def derived[A](using jsonCodec: OFormat[A]): PlayJsonDbCodec[A] = new:
+    def encode(a: A): String = jsonCodec.writes(a).toString
+    def decode(json: String): A = jsonCodec.reads(Json.parse(json)).get
```

Comment:
I wonder if we can't provide these instances thanks to the `Optional` magic of sbt 🤔
See https://blog.7mind.io/no-more-orphans

Reply:
Wow.. that's mind blowing.. thanks for sharing, I'll try it out.

Reply:
Filed https://github.com/AugustNagro/magnum/issues/67

I believe we can merge this as-is and add these instances in a new ticket.